Visual Studio 2017 with /W4: "warning C4127: conditional expression is constant"

I get the following warning message in Visual Studio 2017 with /W4:

```
include/yaml-cpp/node/convert.h(135): warning C4127: conditional expression is constant
include/yaml-cpp/node/convert.h(136): warning C4127: conditional expression is constant
include/yaml-cpp/node/convert.h(137): warning C4127: conditional expression is constant
include/yaml-cpp/node/convert.h(138): warning C4127: conditional expression is constant
include/yaml-cpp/node/convert.h(139): warning C4127: conditional expression is constant
include/yaml-cpp/node/convert.h(140): warning C4127: conditional expression is constant
include/yaml-cpp/node/convert.h(141): warning C4127: conditional expression is constant
include/yaml-cpp/node/convert.h(142): warning C4127: conditional expression is constant
include/yaml-cpp/node/convert.h(144): warning C4127: conditional expression is constant
include/yaml-cpp/node/convert.h(145): warning C4127: conditional expression is constant
include/yaml-cpp/node/convert.h(146): warning C4127: conditional expression is constant
```
yaml-cpp version 0.6.2

--- a/include/yaml-cpp/node/convert.h
+++ b/include/yaml-cpp/node/convert.h
@@ -116,10 +116,11 @@ struct convert<_Null> {
         }                                                                \
       }                                                                  \
                                                                          \
-      if (std::numeric_limits<type>::has_quiet_NaN &&                    \
-          conversion::IsNaN(input)) {                                    \
-        rhs = std::numeric_limits<type>::quiet_NaN();                    \
-        return true;                                                     \
+      if (std::numeric_limits<type>::has_quiet_NaN) {                    \
+        if (conversion::IsNaN(input)) {                                  \
+          rhs = std::numeric_limits<type>::quiet_NaN();                  \
+          return true;                                                   \
+        }                                                                \
       }                                                                  \
                                                                          \
       return false;                                                      \

